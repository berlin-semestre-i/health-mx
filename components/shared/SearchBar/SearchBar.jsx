import React, { PureComponent } from 'react'
import { Input } from 'semantic-ui-react'
import styled from 'styled-components'

const Searchbar = styled(Input)`
  &&& {
    width: 100%;
  }

  &&&>input {
    border-radius: 0;
    height: 70px;
    border: none;
    box-shadow: 0 4px 10px -1px #e3e5e8;
    -webkit-box-shadow: 0 4px 10px -1px #e3e5e8;
  }
`

class SearchBar extends PureComponent {
  render() {
    return(
      <Searchbar icon="search" iconPosition="left" placeholder="Buscar derechohabiente o tarea"/>
    )
  }
}

export default SearchBar