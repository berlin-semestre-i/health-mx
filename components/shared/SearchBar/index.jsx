import React from 'react'
import { Input } from 'semantic-ui-react'
import styled from 'styled-components'

const SearchBar = ({ placeholder }) => (
  <Searchbar icon="search" iconPosition="left" placeholder={placeholder}/>
)

export default SearchBar

const Searchbar = styled(Input)`
  &&& {
    width: 100%;
  }

  &&&>input {
    font-size: 18px;
    font-weight: 300;
    border-radius: 0;
    height: 70px;
    border: none;
    box-shadow: 0 4px 10px -1px #e3e5e8;
    -webkit-box-shadow: 0 4px 10px -1px #e3e5e8;
  }
`
