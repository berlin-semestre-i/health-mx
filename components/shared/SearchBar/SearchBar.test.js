import React from 'react'
import { shallow } from 'enzyme'
import SearchBar from './SearchBar'

describe('SearchBar', () => {
  it('renders', () => {
    const component = shallow(<SearchBar placeholder="test"/>)
    expect(component).toMatchSnapshot()
  })
})