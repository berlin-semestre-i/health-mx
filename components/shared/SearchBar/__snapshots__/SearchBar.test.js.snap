// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    placeholder="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "icon": "search",
      "iconPosition": "left",
      "placeholder": "test",
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "SearchBar__Searchbar-sc-16ll79n-0",
        "isStatic": true,
        "rules": Array [
          "&&&{width:100%;}&&&>input{font-size:18px;font-weight:300;border-radius:0;height:70px;border:none;box-shadow:0 4px 10px -1px #e3e5e8;-webkit-box-shadow:0 4px 10px -1px #e3e5e8;}",
        ],
      },
      "create": [Function],
      "displayName": "SearchBar__Searchbar",
      "handledProps": Array [
        "action",
        "actionPosition",
        "as",
        "children",
        "className",
        "disabled",
        "error",
        "fluid",
        "focus",
        "icon",
        "iconPosition",
        "input",
        "inverted",
        "label",
        "labelPosition",
        "loading",
        "onChange",
        "size",
        "tabIndex",
        "transparent",
        "type",
      ],
      "render": [Function],
      "styledComponentId": "SearchBar__Searchbar-sc-16ll79n-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "icon": "search",
        "iconPosition": "left",
        "placeholder": "test",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "SearchBar__Searchbar-sc-16ll79n-0",
          "isStatic": true,
          "rules": Array [
            "&&&{width:100%;}&&&>input{font-size:18px;font-weight:300;border-radius:0;height:70px;border:none;box-shadow:0 4px 10px -1px #e3e5e8;-webkit-box-shadow:0 4px 10px -1px #e3e5e8;}",
          ],
        },
        "create": [Function],
        "displayName": "SearchBar__Searchbar",
        "handledProps": Array [
          "action",
          "actionPosition",
          "as",
          "children",
          "className",
          "disabled",
          "error",
          "fluid",
          "focus",
          "icon",
          "iconPosition",
          "input",
          "inverted",
          "label",
          "labelPosition",
          "loading",
          "onChange",
          "size",
          "tabIndex",
          "transparent",
          "type",
        ],
        "render": [Function],
        "styledComponentId": "SearchBar__Searchbar-sc-16ll79n-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
